### PR TITLE
chore: bump sphinx to v6.2.1 and myst-parser to v1.0.0

### PR DESCRIPTION
**Pull request by arkodg**
* Renable dependabot updates since the latest modules are compatible

Fixes https://github.com/envoyproxy/gateway/issues/848

### CODE REVIEW

**Review by zirain — APPROVED** [review]

**Comment by zirain** [comment]
I'd like add a label to test the cherrypick action

**Comment by arkodg** [comment]
@zirain does this need @AliceProxy's approval ?

**Comment by zirain** [comment]
it will create a PR, still need approval from maintainer

**Review by Xunzhuo — APPROVED** [review]

**Pull request merged**
